org.scala-ide.sdt.core/src/org/scalaide/core/internal/jdt/model/ScalaSourceFile.scala

```diff
@@ -33,58 +36,60 @@ class ScalaSourceFileProvider extends SourceFileProvider {
 
 object ScalaSourceFile {
 
-  /** Considering [[org.eclipse.jdt.internal.core.util.HandleFactory]] isn't thread-safe, and because
-   *  `ScalaSourceFile#createFromPath` can be called concurrently from different threads, using a
-   *  `ThreadLocal` ensures that a `HandleFactory` instance is never shared across threads.
+  /**
+   * Considering [[org.eclipse.jdt.internal.core.util.HandleFactory]] isn't thread-safe, and because
+   * `ScalaSourceFile#createFromPath` can be called concurrently from different threads, using a
+   * `ThreadLocal` ensures that a `HandleFactory` instance is never shared across threads.
    */
   private val handleFactory: ThreadLocal[HandleFactory] = new ThreadLocal[HandleFactory] {
     override protected def initialValue(): HandleFactory = new HandleFactory
   }
 
-  /** Creates a Scala source file handle if the given resource path points to a scala source.
-   *  The resource path is a path to a Scala source file in the workbench (e.g. /Proj/a/b/c/Foo.scala).
+  /**
+   * Creates a Scala source file handle if the given resource path points to a scala source.
+   * The resource path is a path to a Scala source file in the workbench (e.g. /Proj/a/b/c/Foo.scala).
    *
-   *  @note This assumes that the resource path is the toString() of an `IPath`.
+   * @note This assumes that the resource path is the toString() of an `IPath`.
    *
-   *  @param path Is a path to a Scala source file in the workbench.
+   * @param path Is a path to a Scala source file in the workbench.
    */
-  def createFromPath(path: String) : Option[ScalaSourceFile] = {
+  def createFromPath(path: String): Option[ScalaSourceFile] = {
     if (!path.endsWith(".scala"))
       None
     else {
       // Always `null` because `handleFactory.createOpenable` is only called to open source files, and the `scope` is not needed for this.
       val unusedScope = null
       val source = handleFactory.get().createOpenable(path, unusedScope)
       source match {
-        case ssf : ScalaSourceFile => Some(ssf)
-        case _ => None
+        case ssf: ScalaSourceFile ⇒ Some(ssf)
+        case _                    ⇒ None
```

Comment:
LGTM

2016-08-30 11:47 GMT+02:00 Simon Schäfer <notifications@github.com>:

> In org.scala-ide.sdt.core/src/org/scalaide/core/internal/
> jdt/model/ScalaSourceFile.scala
> <https://github.com/scala-ide/scala-ide/pull/1113#discussion_r76763836>:
>
> >      if (!path.endsWith(".scala"))
> >        None
> >      else {
> >        // Always `null` because `handleFactory.createOpenable` is only called to open source files, and the `scope` is not needed for this.
> >        val unusedScope = null
> >        val source = handleFactory.get().createOpenable(path, unusedScope)
> >        source match {
> > -        case ssf : ScalaSourceFile => Some(ssf)
> > -        case _ => None
> > +        case ssf: ScalaSourceFile ⇒ Some(ssf)
> > +        case _                    ⇒ None
>
> auto format. I don't have anything against => adjusted to the same column
> if it is only one line, that makes it easier to read imho (especially if
> there are more than just two cases).
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/scala-ide/scala-ide/pull/1113/files/382668bacd6d51c6add4e832e28c1b4b64485c58#r76763836>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALEzS4dlx6xwtwgGfP7v0wdI9b7tZag-ks5qk_xAgaJpZM4Juv0C>
> .
>